src/mono-runtimes/mono-runtimes.targets

```diff
@@ -280,26 +278,37 @@
     </ItemGroup>
   </Target>
 
-  <Target Name="_BuildRuntimes"
-      Condition=" '@(_MonoRuntime)' != '' Or '@(_MonoCrossRuntime)' != '' "
-      DependsOnTargets="_GetRuntimesOutputItems"
-      Outputs="@(_RuntimeSource);@(_RuntimeBinarySource);@(_CrossRuntimeBinarySource);@(_ProfilerSource);@(_MonoPosixHelperSource);@(_RuntimeEglibHeaderSource);@(_MonoBtlsSource);@(_BclTestOutput)">
+  <Target Name="_DownloadArchive"
+      DependsOnTargets="_GetMonoGitCommitHash" >
+    <MakeDir
+        Directories="$(MonoSourceFullPath)\sdks\out"
+    />
     <Exec
-        Command="make DISABLE_IOS=1 $(MakeConcurrency) @(_MonoRuntime->'package-android-%(Identity)', ' ') @(_MonoCrossRuntime->'package-android-%(Identity)', ' ') $(_MonoSdksParameters)"
+        Condition=" !Exists('$(MonoSourceFullPath)\sdks\out\.stamp-android-$(_MonoSdksConfiguration)-$(_MonoGitCommitHash)-$(HostOS)-download') "
+        Command="curl --location --silent --show-error https://xamjenkinsartifact.blob.core.windows.net/mono-sdks/android-$(_MonoSdksConfiguration)-$(_MonoGitCommitHash)-$(HostOS).tar.gz | tar -xvf -"
```

Comment:
Instead of `<Exec/>` to `curl`, I would prefer that you use the `<DownloadUri/>` task: https://github.com/xamarin/xamarin-android/blob/master/build-tools/xa-prep-tasks/Xamarin.Android.BuildTools.PrepTasks/DownloadUri.cs

It looks like a point to this PR is to have xamarin-android download the mono-produced bundle directly.  Assuming that's true, then a major benefit to this is that it means that *Windows* should thus be able to build xamarin-android *without a previous macOS xamarin-android build*, which would be *sweet*.  (It would instead require that a prebuilt mono-produced bundle exist.)

Using `<DownloadUri/>` would thus (plausibly) make it possible for Windows to execute this target file.

@jonathanpeppers: Additional thoughts on Windows support?

Reply:
Additional thoughts:

1. The mono bundle should be a `.zip` file, not `.tgz`, so that Windows can plausibly extract the file.
2. `curl | tar xzf -` isn't very resilient to network failures, and could result in a "corrupt" local tree.

Another benefit to using `<DownloadTask/>` is that it ensures that the entire file is properly downloaded before execution continues.

Furthermore, if `<DownloadTask/>` is used, then the downloaded file *should not* be downloaded into `$CWD`.  It should instead be downloaded into `$(AndroidToolchainCacheDirectory)` (usually `$HOME/android-archives`), so that if someone does `git clean -xdff`, the file doesn't need to be re-downloaded again.

See e.g. `build-tools/android-toolchain/android-toolchain.targets`: https://github.com/xamarin/xamarin-android/blob/608d950a703ddb527258565c2a0705aa37f71789/build-tools/android-toolchain/android-toolchain.targets#L55-L58

Reply:
@luhenry ping me when you have this using `DownloadUri` and I will test this on Windows.

In theory, this would be great for our Windows builds, which currently go red whenever we change the mono build, or bump mono.

Reply:
That is not the whole mono-bundle that XA relies on, it only contains the mono bits. I'll try doing it with `<DownloadUri/>`.